### ISSUE TITLE
Problem when restart app on kiosk mode

### ISSUE THREAD

**Issue by hungnq-qest** [Dwrandaz/AutoUpdateComponent#2]
We are developing a POS application which will be setup at our stores. This app is an UWP application which run on Windows 10 IOT LTSC - Kiosk Mode and we decided to deploy using side-loading.

We have used your library to implement the application :)

On desktop mode, It works well with out any problem. After update process, the application will be restarted and the new version will be loaded. But on kiosk mode, sometimes after updated, the application can't restart:

After update process, the application try to restart but it stuck on blue windows with loading spinner.

We also try waiting for a few hours but nothing change

If we try force restart device (use power off button) then after logged on kiosk mode, the application will be loaded with new version.

We have been tested the update process for about 20 times and just 1 times this problem occurred.

If you know about this problem, please help us.

**Comment by mhmd-azeez** [comment]
Hi @hungnq-qest, I am happy that the library has been useful for you. I myself have not used it in kiosk mode, but sometime ago some other person had trouble with it as well. This is what they wrote me:

Email 1:
> I have this integrated into an application I am working and it works probably how you would expect it would - except when I load the app in Assigned Access Mode (Kiosk Mode). When properly into Assigned Access mode, the app ends updating and then getting stuck in a crash loop.

Email 2:
> It looks like my problem is solved. Apparently running the update functionality at app startup time was problematic. After an install/restart is where the app would crash in Assigned Access mode. For troubleshooting purposes I moved the update call behind a button click and found that that works just fine. So I introduced a delay in checking for the update so it does not happen immediately upon the app booting and all appears to be working well now.

I don't know if you're experiencing the same issue, but apparently introducing a little bit of delay in checking for updates solved his problem.

**Comment by hungnq-qest** [comment]
thanks @encrypt0r, I'll try that way to see if that solves the problem :)

**Comment by mhmd-azeez** [comment]
@hungnq-qest Great. Please let me know if it fixes your problem so that I can add the solution to the troubleshoots section of the read me.

**Comment by hungnq-qest** [comment]
We have tried, but the problem still occurred :(

**Comment by mhmd-azeez** [comment]
I see, I am sorry @hungnq-qest I don't know any other fixes for that

**Comment by mykebates** [comment]
Hey all. That was my email copy referenced above. The issue still does come up for us too. It doesn’t happen 100% of the time, though, so trying to track down the issue has proven to be nearly impossible. 

When I sent that email previously it looked like we were in the clear but unfortunately not. Sorry!

**Comment by mhmd-azeez** [comment]
@mykebates Thank you for your input

**Comment by mhmd-azeez** [comment]
Related:
https://docs.microsoft.com/en-us/answers/questions/25947/uwp-problem-when-restart-app-on-kiosk-mode.html

**Comment by FrogsLegs** [comment]
We're facing the same issue, Win 10 LTSC with an Intune kiosk deployment. If we include this package to update our app or even call `CoreApplication.RequestRestartAsync` sometimes the app restarts, and sometimes it doesn't. 

We have isolated this down to the MVVM framework we use; Prism. If we remove Prism from our UWP app, we no longer face this problem. @hungnq-qest @mykebates are you using any kind of MVVM framework within your UWP apps?

**Comment by Berglund81** [comment]
@mykebates  and @hungnq-qest  are you still working on this or have you given up? I have experienced the exact same problem. In my case it feels like the problem disappeard if I create the Kiosk account directly under "Configure Kiosk". I.e. I add a new assigned access acount. I do not first add a user acount and give that assigned access. For some reason this makes a difference for me. No clue why! 
For me this works on the new kiosk account if I use the command: 
`Packagemanager.RequestAddPackageByAppInstallerFileAsync(***);`
It still gets stuck after update when trying to restart if I try to use command: `Packagemanager.UpdatePackageAsync(***);` 
But if I use command `Packagemanager.RequestAddPackageAsync(***)` it works.

**Comment by mykebates** [comment]
@Berglund81 Yeah, we ended up moving away from a native app all together for a number of reasons :(
